packages/iobio-react-components/src/utils/statisticsClient/statisticsTools.mts

```diff
@@ -0,0 +1,119 @@
+/*
+ *
+ *  Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ *  This program and the accompanying materials are made available under the terms of
+ *  the GNU Affero General Public License v3.0. You should have received a copy of the
+ *  GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ *  SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ *  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ *  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ *  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+import { DataBroker } from 'iobio-charts/data_broker.js';
+import fs from 'node:fs';
+import readline from 'node:readline/promises';
+import { calculateMeanCoverage, getBamStatistics } from '../iobioTools.mts';
+import { type StatsOutput } from '../iobioTypes.ts';
+
+export type CompleteCallback = (stats: StatsOutput) => Promise<void>;
+
+/** Generate Iobio metadata using data broker
+ * @param fileUrl Url for the BAM/CRAM file to target
+ * @param fileName Name of target read file, added to output JSON file name
+ * @param indexFileUrl Url for the index file related to the target BAM. Optional
+ * @param enableFileOutput Enable/Disable writing a JSON file with metadata contents
+ * @param onComplete Callback function for when stats-stream finishes. Used to pass stats data to another process. Optional
+ */
+export const generateIobioStats = async ({
+	fileUrl,
+	fileName,
+	indexFileUrl = '',
+	enableFileOutput = false,
+	onComplete,
+}: {
+	fileUrl: string;
+	fileName?: string;
+	indexFileUrl?: string | null;
+	enableFileOutput?: boolean;
+	onComplete?: CompleteCallback | null;
+}) => {
+	// Generate Statistics
+	const serverUrl = process.env.IOBIO_SERVER_URL;
```

Comment:
Environment variables should be handled by code dedicated to reading from the environment and then passed in as an argument. We don't want access to the env scatter throughout our tools.